internal/sync/syncer.go

```diff
@@ -206,8 +207,21 @@ func (s *Syncer) runOnce(ctx context.Context, key string, worker int) error {
 		return fmt.Errorf("failed to query GitSync object of key %q, %w", key, err)
 	}
 	if !gitSync.GetDeletionTimestamp().IsZero() {
-		s.StopWatching(key)
 		log.Debug("GitSync object being deleted.")
+
+		config, err := controllerConfig.GetConfigManagerInstance().GetConfig()
+		if err != nil {
+			log.Errorw("Problem in Getting Config", err)
+			return err
+		}
+		// if cascade deletion is enabled in the config , Delete the linked resources to the GitSync
+		if config.CascadeDeletion {
+			err := CascadeDeletion(ctx, s.client, s.stateCache, gitSync)
+			if err != nil {
+				return err
+			}
+		}
+		s.StopWatching(key)
```

Comment:
all of this code is copy/pasted 3 times. Let's create a method called `ProcessGitSyncDeletion()` or something